packages/api-key/src/services/api-key-module-service.ts

```diff
@@ -329,13 +372,52 @@ export default class ApiKeyModuleService<TEntity extends ApiKey = ApiKey>
     return revokedApiKeys
   }
 
-  // TODO: Implement
   @InjectTransactionManager("baseRepository_")
-  authenticate(
-    id: string,
+  async authenticate(
+    token: string,
     @MedusaContext() sharedContext: Context = {}
-  ): Promise<boolean> {
-    return Promise.resolve(false)
+  ): Promise<ApiKeyTypes.ApiKeyDTO | false> {
+    // Since we only allow up to 2 active tokens, getitng the list and checking each token isn't an issue.
+    // We can always filter on the redacted key if we add support for an arbitrary number of tokens.
+    const secretKeys = await this.apiKeyService_.list(
+      {
+        type: ApiKeyType.SECRET,
+        // If the revoke date is set in the future, it means the key is still valid.
+        $or: [
+          { revoked_at: { $eq: null } },
+          { revoked_at: { $gt: new Date() } },
+        ],
+      },
+      {},
```

Comment:
todo: If we want to get all tokens I would suggest to pass `take: null` to the config options, otherwise the default limit is 15, so you could miss some tokens